<launch>
    <!-- Launch external microphone node -->
    <node pkg="chatting_system" type="external_microphone.py" name="external_microphone" output="screen"/>

    <!-- Launch robot sensors node -->
    <node pkg="chatting_system" type="robot_sensors.py" name="robot_sensors" output="screen"/>

    <!-- robot expression1 -->
    <node pkg="chatting_system" type="robot_head.py" name="robot_head" output="screen"/>

    <!-- Launch sync test node with argument -->
    <arg name="id" default="1"/> <!-- Default ID value can be modified -->
    <node pkg="chatting_system" type="syc_test.py" name="syc_test" output="screen">
        <param name="id" value="$(arg id)"/>
    </node>

    <!-- Launch external tester with argument -->
    <node pkg="chatting_system" type="external_tester.py" name="external_tester" output="screen">
        <param name="id" value="$(arg id)"/>
    </node>

    <!-- Launch GPT4 server with argument -->
    <node pkg="gpt_server" type="gpt4_server.py" name="gpt4_server" output="screen">
        <param name="id" value="$(arg id)"/>
    </node>
</launch>
